sway-core/src/asm_generation/from_ir.rs

```diff
@@ -2287,10 +2396,17 @@ pub fn aggregate_idcs_to_field_layout(
             Type::Union(aggregate) => {
                 let idx = *idx as usize;
                 let field_type = context.aggregates[aggregate.0].field_types()[idx];
+                let union_size_in_bytes = ir_type_size_in_bytes(context, &ty);
                 let field_size_in_bytes = ir_type_size_in_bytes(context, &field_type);
 
-                // The union fields are all at offset 0.
-                ((offs, field_size_in_bytes), field_type)
+                // The union fields are at offset (union_size - variant_size) due to left padding.
```

Comment:
Ahh, OK, not sure if it's the best approach but if it works then it doesn't make that much difference.

Reply:
Argh, catching up on the Slack discussion - they need to be left padded for the ABI.  All good, carry on.